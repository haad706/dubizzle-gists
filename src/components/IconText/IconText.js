import styled from "styled-components";
import PropTypes from "prop-types";
import Octicon from "react-octicon";

export const IconText = ({ iconName, text, link }) => {
  return (
    <Wrapper>
      <Octicon name={iconName} />
      {link ? (
        <AnchorWrapper href={link} target="_blank">
          {text}
        </AnchorWrapper>
      ) : (
        <p>{text}</p>
      )}
    </Wrapper>
  );
};
const Wrapper = styled.div`
  display: inline-flex;
  gap: 0.25rem;
  justify-content: center;
  align-items: center;
  color: #0365d6;
`;

const AnchorWrapper = styled.a`
  text-decoration: none;
  color: #0365d6;
`;

IconText.defaultProps = {
  link: "",
};

IconText.propTypes = {
  iconName: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  link: PropTypes.string,
};

export default IconText;
