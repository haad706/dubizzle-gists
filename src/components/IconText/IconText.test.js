import { render } from "@testing-library/react";
import IconText from "./IconText";

describe("IconText", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<IconText iconName="code" text="Code" />);
    expect(baseElement).toBeTruthy();
  });
  it("should match snapshot", () => {
    const { asFragment } = render(<IconText iconName="code" text="Code" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
