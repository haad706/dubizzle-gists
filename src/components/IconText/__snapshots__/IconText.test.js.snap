// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconText should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bgqQcB ibDLEU"
  >
    <span
      class="octicon octicon-code"
    />
    <p>
      Code
    </p>
  </div>
</DocumentFragment>
`;
